### PR TITLE
Adds SocketTalk for DLLs, Btime for PS 

### PR DESCRIPTION
**Pull request by WhiteFox08**
https://github.com/d3athrow/vgstation13/pull/5640

For some reason  I can't push DLLs in git dispite it not being on the ignore list.

you can find btime here https://dl.dropboxusercontent.com/u/20521740/ShareX/2015/09/btime.dll

### CODE REVIEW

File: code/controllers/master_controller.dm [hunk 1]
```diff
@@ -66,12 +66,23 @@ var/global/pipe_processing_killed = 0
 	//if(!emergency_shuttle)			emergency_shuttle = new /datum/shuttle_controller/emergency_shuttle()
 	if(!supply_shuttle)				supply_shuttle = new /datum/controller/supply_shuttle()
 
+/datum/controller/game_controller/proc/keepalive()
```
Comment (ghost) [inline code/controllers/master_controller.dm]: why would you so this here, and not in a process inside the scheduler?

File: DLLSocket/DLLSocket.cbp [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
```
Comment (ghost) [inline DLLSocket/DLLSocket.cbp]: Don't add tools inside the root of the repository, add them inside the `tools` folder, unless you like a messy codebase.

File: code/controllers/Processes/mob.dm [hunk 3]
```diff
@@ -17,4 +17,4 @@
 /datum/controller/process/mob/doWork()
 	if(updateQueueInstance)
 		updateQueueInstance.init(mob_list, "Life")
```
Comment (ghost) [inline code/controllers/Processes/mob.dm]: updateQueueInstances are removed from the latest process scheduler, change it to a normal loop with proper schecks.

File: code/controllers/configuration.dm [hunk 4]
```diff
@@ -292,13 +293,15 @@
...
+				if("socket_talk")
+					socket_talk = text2num(value)
 				if("tickcomp")
 					Tickcomp = 1
 				if("automute_on")
 					automute_on = 1
 				if("comms_key")
 					global.comms_key = value
-					if(value != "default_pwd" && length(value) > 6) //It's the default value or less than 6 characters long, warn badmins
+					if(value != "default_pwd" && length(value)  > 6) //It's the default value or less than 6 characters long, warn badmins
```
Comment (ghost) [inline code/controllers/configuration.dm:304]: Why the extra space?

File: code/game/socket_talk.dm [hunk 5]
```diff
@@ -0,0 +1,26 @@
+// Module used for fast interprocess communication between BYOND and other processes
+
+/datum/socket_talk
```
Comment (ghost) [inline code/game/socket_talk.dm:3]: no relative pathing, unless you like hard to maintain code.

File: code/controllers/configuration.dm (hunk 4 above)
Comment (WhiteFox08) [inline code/controllers/configuration.dm:304]: because i am anime in real life

**Comment by WhiteFox08** [comment]
Closing this PR there is more stuff I have to fondle.

**Pull request closed without merging**